benchmark/benchmark.go

```diff
@@ -135,9 +152,12 @@ func (s *testServer) UnconstrainedStreamingCall(stream testgrpc.BenchmarkService
 
 	go func() {
 		for {
+			if maxSleep > 0 {
+				time.Sleep(time.Duration(rand.Intn(maxSleep)))
+			}
 			err := stream.Send(response)
 			switch status.Code(err) {
-			case codes.Unavailable:
+			case codes.Unavailable, codes.Canceled:
```

Comment:
I think there is a race here. If server is in the process of sending response while the client closes the connection, we will get Canceled, instead of Unavailable.  Unavailable is still a lot more common. 

This error can be reproduced on master, but with my PR it became a lot more apparent, because now you can increase the number of connections, which greatly increase the chance that at least one connection will result in Canceled.